hack/update-template-alpine.sh

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+# Functions in this script assume error handling with 'set -e'.
+# To ensure 'set -e' works correctly:
+# - Use 'set +e' before assignments and '$(set -e; <function>)' to capture output without exiting on errors.
+# - Avoid calling functions directly in conditions to prevent disabling 'set -e'.
+# - Use 'shopt -s inherit_errexit' (Bash 4.4+) to avoid repeated 'set -e' in all '$(...)'.
+shopt -s inherit_errexit || error_exit "inherit_errexit not supported. Please use bash 4.4 or later."
+
+function alpine_print_help() {
+	cat <<HELP
+$(basename "${BASH_SOURCE[0]}"): Update the Alpine Linux image location in the specified templates
+
+Usage:
+  $(basename "${BASH_SOURCE[0]}") [--version-major-minor <major>.<minor>] <template.yaml>...
+
+Description:
+  This script updates the Alpine Linux image location in the specified templates.
+  Image location basename format:
+
+    <target vendor>_alpine-<version>-<arch>-<firmware>-<bootstrap>[-<machine>]-<image revision>.qcow2
+
+  Published Alpine Linux image information is fetched from the following URLs:
+
+    latest-stable: https://dl-cdn.alpinelinux.org/alpine/latest-stable/releases/cloud
+    <major>.<minor>: https://dl-cdn.alpinelinux.org/alpine/v<major>.<minor>/releases/cloud
+
+  To parsing html, this script requires 'htmlq' or 'pup' command.      
+  The downloaded files will be cached in the Lima cache directory.
+
+Examples:
+  Update the Alpine Linux image location in templates/**.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") templates/**.yaml
+
+  Update the Alpine Linux image location to version 3.18 in ~/.lima/alpine/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") --version-major-minor 3.18 ~/.lima/alpine/lima.yaml
+  $ limactl factory-reset alpine
+
+Flags:
+  --version-major-minor <major>.<minor>  Use the specified <major>.<minor> version.
```

Comment:
What about splitting this to `--version-major` and `--version-minor` for consistency with other scripts

Reply:
I don't think that's a good suggestion. In the future, if I forget and try to specify a version, I can easily see myself getting frustrated with those options.  
But since I probably won't even use this feature, maybe it doesn't matter. 😅

Reply:
added `--version-major` and `--version-minor`